### PR TITLE
Align Vote/Proposal fields with canonical order and fields

### LINKED ISSUE #2727: Align Vote/Proposal fields with canonical order and fields
We should align the Vote/Proposal fields with the Canonicalized fields so eg. Proposal will have SignMsgType and the orders will be the same.

Now that they both are encoded Amino and the order is determined by convenience for HSMs, these should be aligned for consistency so we don't have random orders in the spec.

Ref https://github.com/tendermint/tendermint/issues/2665#issuecomment-433632585

### PR DESCRIPTION
**Pull request by Liamsi**
resolves #2727 

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: types/proposal.go [hunk 1]
```diff
@@ -33,21 +34,27 @@ type Proposal struct {
...
-		POLRound:         polRound,
 		POLBlockID:       polBlockID,
 	}
 }
 
 // String returns a string representation of the Proposal.
 func (p *Proposal) String() string {
-	return fmt.Sprintf("Proposal{%v/%v %v (%v,%v) %X @ %s}",
-		p.Height, p.Round, p.BlockPartsHeader, p.POLRound,
+	return fmt.Sprintf("Proposal{%X(Proposal) %v/%v %v (%v,%v) %X @ %s}",
```
Comment (ebuchman) [inline types/proposal.go]: I don't think we need the `%X(Proposal)` since we already have `Proposal{`

Comment (Liamsi) [reply types/proposal.go]: Thought I make it consistent with Vote but you are right. Removed.

**Comment by Liamsi** [comment]
Updated but there seem to be some problems with circleci pulling the docker image:
```
Error pulling image circleci/golang:1.10.3: Error response from daemon: received unexpected HTTP status: 503 Service Unavailable... retrying
  image is cached as circleci/golang:1.10.3, but refreshing...
```

**Comment by ebuchman** [comment]
Still WIP?

**Comment by ebuchman** [comment]
Oh we need to update spec

**Comment by Liamsi** [comment]
updated the spec. Should I add an entry to the changelog, or changelog pending?

**Comment by ebuchman** [comment]
>  Should I add an entry to the changelog, or changelog pending?

We're in a twilight zone re changelog right now so don't worry about it, I'll handle it on the release front.

Thanks!

File: types/vote.go [hunk 2]
```diff
@@ -85,16 +85,17 @@ func (vote *Vote) String() string {
 		cmn.PanicSanity("Unknown vote type")
 	}
 
-	return fmt.Sprintf("Vote{%v:%X %v/%02d/%v(%v) %X %X @ %s}",
-		vote.ValidatorIndex,
-		cmn.Fingerprint(vote.ValidatorAddress),
-		vote.Height,
-		vote.Round,
+	return fmt.Sprintf("Vote{%v(%v) %v/%02d @ %s %X %v:%X   %X}",
```
Comment (ebuchman) [inline types/vote.go]: Oh I don't think we should change the order in the print statement. It's super convenient for humans to have the validator info at the front

**Pull request merged**